Sources/Valet/SharedGroupIdentifier.swift

```diff
@@ -35,23 +35,47 @@ public struct SharedAccessGroupIdentifier: CustomStringConvertible {
             return nil
         }
 
-        self.appIDPrefix = appIDPrefix
+        self.prefix = appIDPrefix
+        self.groupIdentifier = groupIdentifier
+    }
+
+    /// A representation of a shared app group identifier.
+    /// - Parameters:
+    ///   - groupPrefix: On iOS, iPadOS, watchOS, and tvOS, this prefix must equal "group". On macOS, this prefix is the application's App ID prefix, which  can be found by inspecting the application's provisioning profile, or viewing the application's App ID Configuration on developer.apple.com. This string must not be empty.
+    ///   - groupIdentifier: An identifier that cooresponds to a value in com.apple.security.application-groups in the application's Entitlements file. This string must not be empty.
+    /// - SeeAlso: https://developer.apple.com/documentation/security/keychain_services/keychain_items/sharing_access_to_keychain_items_among_a_collection_of_apps
+    public init?(groupPrefix: String, nonEmptyGroup groupIdentifier: String?) {
```

Comment:
Should we enforce the group prefix on non-macOS by having separate inits?

```swift
#if os(macOS)
public init?(groupPrefix: String, nonEmptyGroup groupIdentifier: String?) {
#else
public init?(nonEmptyGroup groupIdentifier: String?) {
    let groupPrefix = Self.appGroupPrefix
#endif
```

Reply:
We'd need to put the entire initializer in the `#if` (`#if` in Swift is way less permissive than Objective-C), but that'd be doable. I can separate that out.

Reply:
Actually, looking at this more: I think having the `groupPrefix` makes it clear what the `groupIdentifier` should be. If I make a single-parameter initializer for non-macOS platforms, then I think I'd need to accept `groupIdentifier`'s that have `group.` as a prefix, as well as ones that do not. I Kina like this API being explicitly not magical. Thoughts?

Reply:
I think making the `groupIdentifier` only take the non-prefix string is fine as long as we're clear in the headerdoc what the parameter expects (i.e. "this is the identifier following the 'group.' in the entitlement").

I agree with your point that having the `groupPrefix` makes it more obvious, but I think that's outweighed by having a parameter that only accepts a single string. Don't feel strongly though.

Reply:
If we switch to separate initializers, we could also clean up the macOS case by removing the concept of a "group prefix" and specifying that it's the app id prefix.

```swift
init?(nonEmptyGroup groupIdentifier: String?)
init?(appIDPrefix: String, nonEmptyGroup groupIdentifier: String?)
```

Reply:
I like that! But the tough thing is we already have this initializer for shared access groups:
```
init?(appIDPrefix: String, nonEmptyGroup groupIdentifier: String?)
```

There's a good bit of overlap between how these features work. I'm leaning hard towards keeping the APIs explicit for now. We can always add new convenience initializers down the line.